packages/page-accounts/src/modals/CreateEthDerivationPath.tsx

```diff
@@ -0,0 +1,159 @@
+// Copyright 2017-2021 @polkadot/app-accounts authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import BN from 'bn.js';
+import React, { ReactNode, useCallback, useEffect, useRef, useState } from 'react';
+
+import { Checkbox, Dropdown, Input, InputNumber, MarkError, MarkWarning, Modal } from '@polkadot/react-components';
+
+import { useTranslation } from '../translate';
+import { DeriveValidationOutput, PairType } from '../types';
+
+interface Props {
+  className?: string;
+  onChange: (string: string) => void;
+  seedType: string;
+  derivePath: string;
+  deriveValidation: DeriveValidationOutput | undefined;
+  seed: string;
+  addressFromSeed: (seed: string, derivePath: string, pairType: PairType) => string;
+}
+
+export const ETH_DEFAULT_PATH = "m/44'/60'/0'/0/0";
+
+function CreateEthDerivationPath ({ addressFromSeed,
+  className,
+  derivePath,
+  deriveValidation,
+  onChange,
+  seed,
+  seedType }: Props): React.ReactElement<Props> {
+  const { t } = useTranslation();
+  const [addIndex, setAddIndex] = useState(0);
+  const [customIndex, setCustomIndex] = useState(new BN(0));
+  const [addressList, setAddressList] = useState<{ key: number; text: ReactNode; value: number; }[]>([{
+    key: 0,
+    text: t('Address index 0'),
+    value: 0
+  }]);
+  const [useCustomPath, setUseCustomPath] = useState(false);
+  const [useCustomIndex, setUseCustomIndex] = useState(false);
+
+  const errorIndex = useRef<Record<string, string>>({
+    INVALID_DERIVATION_PATH: t<string>('This is an invalid derivation path.'),
+    PASSWORD_IGNORED: t<string>('Password are ignored for hex seed'),
+    SOFT_NOT_ALLOWED: t<string>('Soft derivation paths are not allowed on ed25519'),
+    WARNING_SLASH_PASSWORD: t<string>(
+      'Your password contains at least one "/" character. Disregard this warning if it is intended.'
+    )
+  });
+
+  const _toggleCustomPath = useCallback(() => {
+    setUseCustomPath(!useCustomPath);
+  }, [setUseCustomPath, useCustomPath]);
+
+  const _toggleCustomIndex = useCallback(() => {
+    setUseCustomIndex(!useCustomIndex);
+  }, [useCustomIndex, setUseCustomIndex]);
+
+  const _onChangeCustomIndex = useCallback((e: BN) => {
+    setCustomIndex(e);
+  }, [setCustomIndex]);
+
+  useEffect((): void => {
+    if (!useCustomIndex) {
+      console.log(1);
+      onChange(`m/44'/60'/0'/0/${addIndex}`);
+    } else {
+      console.log(2);
+      onChange(`m/44'/60'/0'/0/${Number(customIndex)}`);
+    }
+  }, [customIndex, onChange, useCustomIndex, addIndex]);
+
+  useEffect((): void => {
+    const list = new Array(5).fill(0).map((_, i) => ({
+      key: i,
+      text: t('Address index {{index}} - {{address}}', {
+        replace: { address: addressFromSeed(seed, `m/44'/60'/0'/0/${i}`, 'ethereum'), index: i }
+      }),
+      value: i
+    }));
+
+    setAddressList(list);
+  }, [seed, setAddressList, addressFromSeed, t]);
```

Comment:
Since we just effect and set, this is probably better suited with a `useMemo`? i.e.

```js
const addressList = useMemo(() => {
...
}, [seed, addressFromSeed, t]);
```

Having said that, the `addressFromSeed` is quite expensive, so not sure we want to calculate that many. Would suggest rather on index change to do 1. (With the derivation there is a noticeable delay that will happen here in the main thread)

Reply:
>Would suggest rather on index change to do 1.

Not sure what you mean

Reply:
So we are creating 5 entries - the entries are not the issue, deriving from the seed and making an address is horrible.

So the option is to not derive 5 addresses - only setup the indexes, create the address only for the current selected index - as it changes.

Reply:
The ability to see multiple addresses in the dropdown is a feature that our Developer Relations Engineer requested in order to mimic the 'standard' ux for hd derivation as seen in MetaMask or ganache. On my computer it loads fast enough. If you think this feature is too ressource intensive I can circle back to my DevRel colleague to see if we can reach a compromise

Reply:
I'll merge with it in, but I am going to rip it out :) It really is not a great idea unless it is done in a worker.

Reply:
I don't understand what you are saying. I don't want you to merge something that you will take out. If you don't want this feature in your app, say it and I will remove it.

Reply:
I've mentioned a couple of time now that it is an exceptionally bad idea, that should not be deployed :)

Reply:
Ok I understand. Since I'm not used to webworkers and this is not high priority I'm just going to remove the address generation in the dropdown